tests/rptest/tests/schema_registry_test.py

```diff
@@ -1264,11 +1271,13 @@ def test_schema_id_validation(self,
                                   validate_schema_id: Optional[bool] = None,
                                   subject_name_strategy: Optional[str] = None,
                                   payload_class: str = "com.redpanda.Payload"):
-        Admin(self.redpanda).put_feature("schema_id_validation",
-                                         {"state": "active"})
-        self.redpanda.await_feature("schema_id_validation",
-                                    True,
-                                    timeout_sec=10)
+        admin = Admin(self.redpanda)
+        patch_result = admin.patch_cluster_config(
+            upsert={
+                'enable_schema_id_validation': SchemaIdValidationMode.COMPAT
+            })
+        wait_for_version_status_sync(admin, self.redpanda,
+                                     patch_result['config_version'])
```

Comment:
Probably. Is there a particular concern in changing the configuration at the start of the test?

I'll change it to the slightly more readable:
```python
        self.redpanda.set_cluster_config({'enable_schema_id_validation': mode})
```

Reply:
> Probably. Is there a particular concern in changing the configuration at the start of the test?

No concern really, these lines just seemed like they could go somewhere else instead of this test.

> I'll change it to the slightly more readable:
> 
> ```python
>         self.redpanda.set_cluster_config({'enable_schema_id_validation': mode})
> ```

That will do fine for readability purposes.